Toolset/palettes/inspector/editors/com.livecode.pi.gradient.behavior.livecodescript

```diff
@@ -0,0 +1,110 @@
+﻿script "com.livecode.pi.gradient.behavior"
+local sPopupDefaults
+
+on editorInitialize
+   set the editorMinWidth of me to 25
+   set the editorMaxWidth of me to 0
+   
+   local tPopupData, tDefaultStore
+   put editorPropertyGet("popup") into tPopupData
+   repeat for each element tGroup in tPopupData
+      repeat for each element tProp in tGroup["proplist"]
+         put tProp["default"] into sPopupDefaults[tProp["property_name"]]
+         put tProp["default"] into tDefaultStore[char 9 to -1 of tProp["property_name"]]
+      end repeat
+   end repeat
+   
+   if the storedValue of me is empty then
+      set the editorValue of me to tDefaultStore
+      storeProperty
+      set the editorValue of me to empty
+   end if
+end editorInitialize
+
+private function firstNonEmpty pLeft, pRight
+   if pLeft is empty then
+      return pRight
+   end if
+   return pLeft
+end firstNonEmpty
```

Comment:
I think this can be removed, I used to use it in the graphic fx editor but maybe redundant there too